omnipaxos/src/sequence_paxos/mod.rs

```diff
@@ -272,40 +274,36 @@ where
         }
     }
 
-    /// Propose a reconfiguration. Returns error if already stopped or new configuration is empty.
-    pub(crate) fn reconfigure(&mut self, rc: ReconfigurationRequest) -> Result<(), ProposeErr<T>> {
-        let ReconfigurationRequest {
-            new_configuration,
-            metadata,
-        } = rc;
+    /// Propose a reconfiguration. Returns error if already stopped.
+    pub(crate) fn reconfigure(&mut self, new_config: ClusterConfig) -> Result<(), ProposeErr<T>> {
         #[cfg(feature = "logging")]
         info!(
             self.logger,
-            "Propose reconfiguration {:?}", new_configuration
+            "Propose reconfiguration {:?}", new_config.nodes
         );
         if self.stopped() {
-            Err(ProposeErr::Reconfiguration(new_configuration))
+            Err(ProposeErr::Reconfiguration(new_config))
         } else {
             match self.state {
                 (Role::Leader, Phase::Prepare) => {
                     if self.pending_stopsign.is_none() {
-                        let ss = StopSign::with(self.config_id + 1, new_configuration, metadata);
```

Comment:
It would be nice to still be able to have some user-provided metadata that gets decided with the reconfiguration. Probably it is better to have it as a second argument for `reconfigure()`.